api/xds_protocol.rst

```diff
@@ -36,43 +36,122 @@ occurs.
 
 The concept of `type URLs <https://developers.google.com/protocol-buffers/docs/proto3#any>`_ appears below, and takes the form
-`type.googleapis.com/<resource type>`, e.g.
-`type.googleapis.com/envoy.api.v2.Cluster` for CDS. In various
-requests from Envoy and responses by the management server, the resource
-type URL is stated.
+`type.googleapis.com/<resource type>` -- e.g., `type.googleapis.com/envoy.api.v2.Cluster` for a `Cluster` resource. In various
+requests from Envoy and responses by the management server, the resource type URL is stated.
+
+Variants of the xDS API
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Four Variants
+^^^^^^^^^^^^^
+
+There are four variants of the xDS API used via streaming gRPC, which cover all combinations of two dimensions.
+
+The first dimension is State of the World (SotW) vs. incremental. The SotW approach was the original mechanism used by xDS, in
+which the client must specify all resource names it is interested in with each request (except when making a wildcard
+request in LDS/CDS), and the server must return all resources the client has subscribed to in each request (in LDS/CDS).
+This means that if the client is already subscribing to 99 resources and wants to add an additional one, it must send a request
+with all 100 resource names, rather than just the one new one. And the server must then respond by sending all 100 resources,
+even if the 99 that were already subscribed to have not changed (in LDS/CDS). This mechanism can be a scalability limitation,
+which is why the incremental API variant was introduced. The incremental approach allows both the client and server to indicate
+only deltas relative to their previous state -- i.e., the client can say that it wants to add or remove its subscription to a
+particular resource name without resending those that have not changed, and the server can send updates only for those
+resources that have changed. For details on the incremental protocol, see :ref:`Incremental xDS <xds_protocol_incremental>`
+below.
+
+The second dimension is using a separate gRPC stream for each resource type vs. aggregating all resource types onto a
+single gRPC stream. The former approach was the original mechanism used by xDS, and it offers an eventual consistency
+model. The latter approach was added for environments in which explicit control of sequencing is required. For details, see
+:ref:`Eventual consistency considerations <xds_protocol_eventual_consistency>` below.
+
+So, the four variants of the xDS API are:
+
+1. State of the World (Basic xDS): SotW, separate gRPC stream for each resource type
+2. Incremental xDS: incremental, separate gRPC stream for each resource type
+3. Aggregated Discovery Service (ADS): SotW, aggregate stream for all resource types
+4. Incremental ADS: incremental, aggregate stream for all resource types
+
+In general, new xDS implementations are encouraged to use the last variant.
```

Comment:
I think this is stronger statement than we've made to date; we try and support eventual consistent with multiple streams as first class, and will continue to do so in UDPA. CC @mattklein123

Reply:
I'd just drop the recommendation for now. I agree with what you write above, but I don't think we have any prescriptive stance today on which one folks should be using. I think that as long as v2/v3 xDS exists, we're going to fully support all variants and will want to allow management server implementors the option to pick.